### PR TITLE
[PLATFORM-565] feat: Add new resources commands

### PR DESCRIPTION
**Pull request by raulb**
# Description of change

Fixes https://meroxa.atlassian.net/browse/PLATFORM-565.

This pull request contains all commands related to resources while also migrating them to use the new builder process adding tests to some commands that didn't have them before.

- [x] create
- [x] describe
- [x] list
- [x] list --types (it used to be list resource-types)
- [x] remove
- [x] update
 
The idea to ultimately release this will be with the use of `MEROXA_V2` env variable.

example:

- `export MEROXA_V2=only` would make only v2 commands to be available
- `unset MEROXA_V2` would leave the CLI as v1 only.
- `export MEROXA_V2=any_value` would make both v1 and v2 commands coexist. We could decide to remove this restriction once we're feature complete on v2.

# Type of change

- [x]  New feature
- [ ]  Bug fix
- [x]  Refactor
- [ ]  Documentation

# How was this tested?

- [x]  Unit Tests
- [ ]  Tested in staging

# Demo

**Before this pull-request**

```
❯ .m help
The Meroxa CLI allows quick and easy access to the Meroxa data platform.

Using the CLI you are able to create and manage sophisticated data pipelines
with only a few simple commands. You can get started by listing the supported
resource types:

meroxa list resource-types

Usage:
  meroxa [command]

Available Commands:
  add         Add a resource to your Meroxa resource catalog
  api         Invoke Meroxa API
  billing     Open your billing page in a web browser
  completion  Generate completion script
  connect     Connect two resources together
  create      Create Meroxa pipeline components
  describe    Describe a component
  help        Help about any command
  list        List components
  login       login or sign up to the Meroxa platform
  logout      logout of the Meroxa platform
  logs        Print logs for a component
  open        Open in a web browser
  remove      Remove a component
  update      Update a component
  version     Display the Meroxa CLI version
  whoami      Display the current logged in user


Flags:
      --config string      config file
      --debug              display any debugging information
  -h, --help               help for meroxa
      --json               output json
      --timeout duration   set the client timeout (default 10s)

Use "meroxa [command] --help" for more information about a command
```

**After this pull-request (with `MEROXA_V2=only`)**

```
❯ .m help resources
Manage resources on Meroxa

Usage:
  meroxa resources [command]

Aliases:
  resources, resource

Available Commands:
  create      Add a resource to your Meroxa resource catalog
  describe    Describe resource
  list        List resources and resource types
  remove      Remove resource
  update      Update a resource

Flags:
  -h, --help   help for resources

Global Flags:
      --config string      config file
      --debug              display any debugging information
      --json               output json
      --timeout duration   set the client timeout (default 10s)

Use "meroxa resources [command] --help" for more information about a command.

❯ .m help resources create
Use the create command to add resources to your Meroxa resource catalog.

Usage:
  meroxa resources create [NAME] --type TYPE --url URL [flags]

Aliases:
  create, add

Examples:

meroxa resources create store --type postgres -u $DATABASE_URL --metadata '{"logical_replication":true}'
meroxa resources create datalake --type s3 -u "s3://$AWS_ACCESS_KEY_ID:$AWS_ACCESS_KEY_SECRET@us-east-1/meroxa-demos"
meroxa resources create warehouse --type redshift -u $REDSHIFT_URL
meroxa resources create slack --type url -u $WEBHOOK_URL


Flags:
      --ca-cert string       trusted certificates for verifying resource
      --client-cert string   client certificate for authenticating to the resource
      --client-key string    client private key for authenticating to the resource
  -h, --help                 help for create
  -m, --metadata string      resource metadata
      --password string      password
      --ssl                  use SSL
      --type string          resource type
  -u, --url string           resource url
      --username string      username

Global Flags:
      --config string      config file
      --debug              display any debugging information
      --json               output json
      --timeout duration   set the client timeout (default 10s)

❯ .m help resources describe
Describe resource

Usage:
  meroxa resources describe [NAME] [flags]

Flags:
  -h, --help   help for describe

Global Flags:
      --config string      config file
      --debug              display any debugging information
      --json               output json
      --timeout duration   set the client timeout (default 10s)

❯ .m help resources list
List resources and resource types

Usage:
  meroxa resources list [flags]

Aliases:
  list, ls

Flags:
  -h, --help    help for list
      --types   list resource types

Global Flags:
      --config string      config file
      --debug              display any debugging information
      --json               output json
      --timeout duration   set the client timeout (default 10s)

❯ .m help resources remove
Remove resource

Usage:
  meroxa resources remove NAME [flags]

Aliases:
  remove, rm, delete

Flags:
  -f, --force   skip confirmation
  -h, --help    help for remove

Global Flags:
      --config string      config file
      --debug              display any debugging information
      --json               output json
      --timeout duration   set the client timeout (default 10s)

❯ .m help resources update
Use the update command to update various Meroxa resources.

Usage:
  meroxa resources update NAME [flags]

Flags:
      --ca-cert string       trusted certificates for verifying resource
      --client-cert string   client certificate for authenticating to the resource
      --client-key string    client private key for authenticating to the resource
  -h, --help                 help for update
  -m, --metadata string      new resource metadata
      --name string          new resource name
      --password string      password
      --ssl                  use SSL
  -u, --url string           new resource url
      --username string      username

Global Flags:
      --config string      config file
      --debug              display any debugging information
      --json               output json
      --timeout duration   set the client timeout (default 10s)
```

# Additional references

[Design document](https://www.notion.so/meroxa/CLI-subject-verb-object-fd84c27e3a2444638172f5f892f1b2ce)

### CODE REVIEW

File: cmd/meroxa/root/resources/list.go [hunk 1]
```diff
@@ -0,0 +1,106 @@
...
+}
+
+func (l *List) Aliases() []string {
+	return []string{"ls"}
+}
+
+func (l *List) Execute(ctx context.Context) error {
+	var err error
+
+	// What used to be `meroxa list resource-types`
```
Comment (lovromazgon) [inline cmd/meroxa/root/resources/list.go:73]: This comment will soon be deprecated.

**Review by lovromazgon — APPROVED** [review]
Some non-blocking comments. LGTM overall!

File: cmd/meroxa/root/resources/update.go [hunk 2]
```diff
@@ -0,0 +1,186 @@
...
+
+func (u *Update) setMetadata(res *meroxa.UpdateResourceInput) error {
+	var metadata map[string]interface{}
+	err := json.Unmarshal([]byte(u.flags.Metadata), &metadata)
+	res.Metadata = metadata
+	return err
+}
+
+func (u *Update) setCredentials(res *meroxa.UpdateResourceInput) {
+	res.Credentials = &meroxa.Credentials{}
```
[the remark this answers is not in the document]
Comment (raulb) [reply cmd/meroxa/root/resources/update.go:161]: Addressed in https://github.com/meroxa/cli/commit/36f69e83f4b4519e1fc2cd51db784010f50b483a

File: cmd/meroxa/root/resources/create.go [hunk 3]
```diff
@@ -0,0 +1,152 @@
...
+			ClientCert:    c.flags.ClientCert,
+			ClientCertKey: c.flags.ClientKey,
+			UseSSL:        c.flags.SSL,
+		}
+	}
+
+	if c.flags.Metadata != "" {
+		err := json.Unmarshal([]byte(c.flags.Metadata), &input.Metadata)
+		if err != nil {
+			return err
```
[the remark this answers is not in the document]
Comment (raulb) [reply cmd/meroxa/root/resources/create.go:128]: That's a great idea @lovromazgon

[the remark this answers is not in the document]
Comment (raulb) [reply cmd/meroxa/root/resources/create.go:128]: Partially addressed in https://github.com/meroxa/cli/commit/1b04fd35e0469adc9a61ff79f8c888c2de2369e6.

**Pull request merged**